scripts/lint-mdc.ts

```diff
@@ -496,28 +757,234 @@ function checkOneClassMemberIsInReadme(
     logLinterViolation(
         inputFilePath,
         loc,
-        `Public ${kind} '${name}' is missing from '${readmeFilePathRelative}'.`,
+        [
+          `Missing documentation for ${kind}: '${name}'.`,
+          `Most publicly exported symbols must be documented in '${readmeFilePathRelative}'.`,
+        ],
     );
     return;
   }
 }
 
-function getNameAndLocation(node: babelTypes.Node): NamedIdentifier {
-  if (babelTypes.isClassMethod(node) || babelTypes.isClassProperty(node) ||
-      babelTypes.isObjectMethod(node) || babelTypes.isObjectProperty(node) ||
+function checkAllAdaptersAndFoundations(inputFilePath: string, inputCode: string) {
```

Comment:
Would this actually make more sense to be 2 separate functions, one for adapters, one for foundations? I'm honestly finding it confusing why there's an extra traversal for the adapter case anyway.